src/components/CowPen/CowPen.js

```diff
@@ -1,34 +1,39 @@
 import React, { Component } from 'react'
 import { array, bool, func, object, string } from 'prop-types'
 import classNames from 'classnames'
+import { Tweenable } from 'shifty'
 import Tooltip from '@material-ui/core/Tooltip'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faHeart } from '@fortawesome/free-solid-svg-icons'
 
 import { cowColors } from '../../enums'
+import { LEFT, RIGHT } from '../../constants'
 import FarmhandContext from '../../Farmhand.context'
 import { animals } from '../../img'
 
 import './CowPen.sass'
 
 export class Cow extends Component {
   state = {
-    isMoving: false,
+    isTransitioning: false,
+    moveDirection: RIGHT,
+    rotate: 0,
     showHugAnimation: false,
     x: Cow.randomPosition(),
     y: Cow.randomPosition(),
   }
 
   repositionTimeoutId = null
-  animateMovementTimeoutId = null
   animateHugTimeoutId = null
+  tweenable = new Tweenable()
 
-  // This MUST be kept in sync with $movement-animation-duration in CowPen.sass.
-  static movementAnimationDuration = 3000
+  static flipAnimationDuration = 1000
```

Comment:
we can move to native CSS variables and use `getComputedStyle` if you would like to remove this need to update code in 2 places. alternatively, the javascript can be used to seed the CSS variable valuable depending on which one you feel is easier to maintain (not intending this comment holds up this PR, but it's just some food for thought)

Reply:
I'm open minded here! I think the variables in question are inherently a display concern, so I'd prefer the source of truth live in a Sass/CSS file.